Pod/Classes/EigenCommunications/ARNotificationsManager.m

```diff
@@ -0,0 +1,26 @@
+#import "ARNotificationsManager.h"
+
+@implementation ARNotificationsManager
+
+RCT_EXPORT_MODULE();
+
+- (instancetype)init;
+{
+    if ((self = [super init])) {
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(paymentRequestPaidReceived:) name:@"ARPaymentRequestPaid" object:nil];
+    }
+    return self;
+}
+
+- (NSArray<NSString *> *)supportedEvents
+{
+    return @[@"PaymentRequestPaid"];
+}
+
+- (void)paymentRequestPaidReceived:(NSNotification *)notification
+{
+    NSDictionary *info = notification.userInfo;
+    [self sendEventWithName:@"PaymentRequestPaid" body:@{@"url": info[@"ARPaymentRequestURL"] }];
```

Comment:
There are two things still missing here. One is to _not_ forward events over the bridge when there are no resgitered listeners. The second, which we didn’t discuss previously, is to unregister from the `NSNotificationCenter` when this native module is released (i.e. when the bridge is released, which doesn’t actually happen in Eigen right now, but best to add it now so we never forget).

```objc
// This is the instance method that’s invoked on an object when it’s about to be released. (Don’t call super, the compiler will complain about it too.)
- (void)dealloc;
{
  [[NSNotificationCenter defaultCenter] removeObserver:self];
}
```